File: frontend/src/microservices/tour-group/objects/TourGroupConversationSummaryResponse.ts
// 本文件定义 tour-group 模块的 `TourGroupConversationSummaryResponse`，作为摘要响应数据并提供 JSON 编解码。

export type TourGroupConversationSummaryResponse = {
  conversationId: string
  conversationType: string
  status: string
  counterpartUserId: string | null
  counterpartDisplayName: string | null
  counterpartAvatarUrl: string | null
  conversationTitle: string
  participantsSummary: string
  lastMessagePreview: string | null
  lastMessageAt: string | null
  unreadCount: number
  isMuted: boolean
  isArchived: boolean
  canSendMessage: boolean
}
export const tourGroupConversationSummaryResponseFromJson = (json: string): TourGroupConversationSummaryResponse =>
  JSON.parse(json) as TourGroupConversationSummaryResponse

export const tourGroupConversationSummaryResponseToJson = (value: TourGroupConversationSummaryResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupConversationSummaryResponse.scala
// TourGroupChatPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import com.typesafe.travel.shared.kernel.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

import java.time.Instant

final case class TourGroupConversationSummaryResponse(
    conversationId: String,
    conversationType: String,
    status: String,
    counterpartUserId: Option[String],
    counterpartDisplayName: Option[String],
    counterpartAvatarUrl: Option[String],
    conversationTitle: String,
    participantsSummary: String,
    lastMessagePreview: Option[String],
    lastMessageAt: Option[String],
    unreadCount: Int,
    isMuted: Boolean,
    isArchived: Boolean,
    canSendMessage: Boolean
)
object TourGroupConversationSummaryResponse:
  given sourceEncoder: Encoder[TourGroupConversationSummaryResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupConversationSummaryResponse] = deriveDecoder
